java-extensions/udf-extensions/src/main/java/com/starrocks/udf/CallStubGenerator.java

```diff
@@ -0,0 +1,146 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+package com.starrocks.udf;
+
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Type;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+
+import static org.objectweb.asm.Opcodes.*;
+
+public class CallStubGenerator {
+    // generate call stub name
+    public static final String CLAZZ_NAME = "com/starrocks/udf/gen/CallStub";
+    public static final String GEN_KEYWORD = "com.starrocks.udf.gen";
+
+    // generate batch update
+    // public class CallStub {
+    //     public static void batchCallV(int rows, UDAFSum obj, State var0, Integer[] var1, ...) throws Exception {
+    //         for(int i = 0; i < rows; ++i) {
+    //             obj.update(var0, var1[i], ...);
+    //         }
+    //     }
+    // }
+    private static class AggBatchCallGenerator {
+        AggBatchCallGenerator(Class<?> clazz, Method update) {
+            this.UDAFClazz = clazz;
+            this.UDAFUpdate = update;
+        }
+
+        private final Class<?> UDAFClazz;
+        private final Method UDAFUpdate;
+
+        private final ClassWriter writer = new ClassWriter(0);
+
+        private void declareCallStubClazz() {
+            writer.visit(V1_8, ACC_PUBLIC, CLAZZ_NAME, null, "java/lang/Object", null);
+        }
+
+        // int numRows, FunctionCallClz obj, FunctionCall.State state, Integer[] a
+        private void genBatchUpdateSingle() {
```

Comment:
These codes are hard to maintain

Reply:
we could provide some wrapper like presto